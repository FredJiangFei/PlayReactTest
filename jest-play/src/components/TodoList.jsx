import React from 'react';
import Todo from './Todo';

function TodoList({ items }) {
  return (
    <div>
      <h3>Lists</h3>
      {!items.length ? (
        <p>No task!</p>
      ) : (
        <ul>
          {items.map((item) => {
            return (
              <li key={item.id}>
                <Todo {...item} />
              </li>
            );
          })}
        </ul>
      )}
    </div>
  );
}

export default TodoList;
