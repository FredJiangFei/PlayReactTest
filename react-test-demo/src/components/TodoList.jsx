import React from 'react'
import TodoItem from './TodoItem'

function TodoList({ items, onRemoveTodo }) {
  return (
    <div>
      <h3>Lists</h3>
      {!items.length ? (
        <p className='no-task'>No task!</p>
      ) : (
        <ul>
          {items.map((item) => {
            return (
              <li key={item.id}>
                <TodoItem {...item} onRemoveTodo={onRemoveTodo}/>
              </li>
            )
          })}
        </ul>
      )}
    </div>
  )
}

export default TodoList
