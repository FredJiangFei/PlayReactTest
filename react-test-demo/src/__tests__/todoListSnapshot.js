import { render } from '@testing-library/react';
import React from 'react';
import TodoList from './../components/TodoList';

test('renders the todos empty snapshot', () => {
  // arrange
  var items = [];

  // act
  const { container } = render(<TodoList items={items} />);

  // assert
  expect(container.firstChild).toMatchSnapshot();
});

test('renders the todos snapshot', () => {
  // arrange
  var items = [
    {
      id: 1,
      title: 'delectus aut autem'
    },
    {
      id: 2,
      title: 'quis ut nam facilis et officia qui'
    },
  ];

  // act
  const { container } = render(<TodoList items={items} />);

  // assert
  expect(container.firstChild).toMatchSnapshot();
});

it('renders the users snapshot with id and date', () => {
  const user = {
    createdAt: new Date(),
    id: Math.floor(Math.random() * 20),
    name: 'LeBron James',
  };

  expect(user).toMatchSnapshot({
    createdAt: expect.any(Date),
    id: expect.any(Number),
  });
});
