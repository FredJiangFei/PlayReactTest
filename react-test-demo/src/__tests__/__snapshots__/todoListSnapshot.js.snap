// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the todos empty snapshot 1`] = `
<div>
  <h3>
    Lists
  </h3>
  <p>
    No task!
  </p>
</div>
`;

exports[`renders the todos snapshot 1`] = `
<div>
  <h3>
    Lists
  </h3>
  <ul>
    <li>
      delectus aut autem
       
      <button>
        X
      </button>
    </li>
    <li>
      quis ut nam facilis et officia qui
       
      <button>
        X
      </button>
    </li>
  </ul>
</div>
`;

exports[`renders the users snapshot with id and date 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": "LeBron James",
}
`;
